common/protos/index.ts

```diff
@@ -14,6 +14,33 @@ export interface IProtoClass<IProto, Proto> {
   fromObject(obj: { [k: string]: any }): Proto;
 }
 
+// ProtobufJS's native verify method does not check that only defined fields are present.
+// TODO(ekrekr): swap to Typescript protobuf library rather than having to do this.
+export function verifyObjectMatchesProto<Proto>(
```

Comment:
i hate this, let's please switch

Reply:
I did some digging into this, and the alternatives which can validate (https://www.npmjs.com/package/@protobuf-ts/plugin, https://www.npmjs.com/package/protoc-gen-ts) are significantly slower. More details in b/305915462#comment18.